File: Practice/IntentExample/app/src/main/java/com/example/intentexample/MainActivity.java
package com.example.intentexample;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    private Button button;
    private EditText text;
    private String str;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        text = (EditText) findViewById(R.id.text);

        button = (Button) findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                str = text.getText().toString();
                Intent intent = new Intent(MainActivity.this, SubActivity.class);
                intent.putExtra("str", str);
                startActivity(intent);
            }
        });

    }
}

File: SOPT_Seminar2/Assignment/S2_A2/app/src/main/java/com/example/s2_a2/MainActivity.kt
package com.example.s2_a2

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var dogList = arrayListOf<Dog>(
        Dog("Golden Retreiver", "Female", "2", "dog1"),
        Dog("Pug", "Male", "4", "dog2"),
        Dog("Huskey", "Male", "3", "dog3"),
        Dog("Poodle", "Female", "5", "dog4"),
        Dog("Yorkshire Terrier", "Male", "1", "dog5"),
        Dog("Bedlington Terrier", "Female", "3", "dog6")
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val mAdapter = MainRvAdapter(this, dogList)
        mRecyclerView.adapter = mAdapter
        mRecyclerView.addItemDecoration(CustomDividerItemDecoration(this))

        val lm = LinearLayoutManager(this)
        mRecyclerView.layoutManager = lm
        mRecyclerView.setHasFixedSize(true)
    }
}
